### PR TITLE
Added unicode_literals to GenericProvider

### PR DESCRIPTION
**Pull request by p0psicles**
- [ ] PR is based on the DEVELOP branch
- [ ] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [ ] Read [contribution guide](https://github.com/PyMedusa/SickRage/blob/master/.github/CONTRIBUTING.md)

* Also adapted all providers, to make use of the future import unicode_literals
* Removed the decode()/encode()
* Cleaned up some double to single quotes
* Added proper exceptions for the provider results items
* Some logging cleanup using format()

### CODE REVIEW

File: sickbeard/providers/newznab.py [hunk 1]
```diff
@@ -125,7 +125,6 @@ def get_providers_list(data):
                 providers_dict[default.name].enable_daily = default.enable_daily
                 providers_dict[default.name].enable_backlog = default.enable_backlog
                 providers_dict[default.name].enable_manualsearch = default.enable_manualsearch
-                providers_dict[default.name].catIDs = default.catIDs
```
Comment (p0psicles) [inline sickbeard/providers/newznab.py]: This was the cause for an issue where the category id's are always overwritten, after a restart. You could change the categories in the provider_config. But a restart would revert it back to 5030,5040

File: sickbeard/providers/newznab.py [hunk 2]
```diff
@@ -276,12 +275,11 @@ def search(self, search_strings, age=0, ep_obj=None):  # pylint: disable=too-man
         if not self._check_auth():
             return results
 
-        # gingadaddy has no caps.
-        if not self.caps and 'gingadaddy' not in self.url:
+        # For providers that don't have no caps, or for which the t=caps is not working.
+        if not self.caps and all(provider not in self.url for provider in ['gingadaddy', 'usenet-crawler']):
```
Comment (p0psicles) [inline sickbeard/providers/newznab.py]: usenet-crawler gives back a 500 error, when trying to get the category caps.

File: sickbeard/providers/tntvillage.py [hunk 3]
```diff
@@ -170,37 +172,37 @@ def _reverseQuality(quality):
 
     @staticmethod
     def _episodeQuality(torrent_rows):  # pylint: disable=too-many-return-statements, too-many-branches
-        """
+        '''
```
Comment (p0psicles) [inline sickbeard/providers/tntvillage.py]: fck, made the same mistake. Will correct that

**Comment by medariox** [comment]
Approved

**Comment by p0psicles** [comment]
Approved

**Pull request merged**